### PR TITLE
Make jib build

### PR DESCRIPTION
**Pull request by rafaelraposospot**
# TL;DR
Make Jib upload build to Github Container Registry

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

**Review by andresgomezfrr — APPROVED** [review]

**Review by narape — APPROVED** [review]

**Comment by honnix** [comment]
Does this work? I don't see docker push happening during release or main build. https://github.com/flyteorg/flytekit-java/blob/8076468ed22180e75adef19da74769631b9ca6c7/jflyte/pom.xml#LL242C21-L242C21 here we only bind to `deploy` phase.
